core/state/trie_prefetcher.go

```diff
@@ -172,13 +237,10 @@ func (p *triePrefetcher) prefetch(root common.Hash, keys [][]byte, accountHash c
 	if p.fetches != nil {
 		return
 	}
-	// Active fetcher, schedule the retrievals
-	fetcher := p.fetchers[root]
-	if fetcher == nil {
-		fetcher = newSubfetcher(p.db, root, accountHash)
-		p.fetchers[root] = fetcher
+	select {
+	case <-p.closeMainChan: // skip closed trie prefetcher
+	case p.prefetchChan <- &prefetchMsg{root, accountHash, keys}:
```

Comment:
I will suggest add a `case default` here. If the `p.prefetchChan` if full, it will block the transaction execution.

Reply:
will skip since no `for` loop here. `select` will keep try the channel till one is available, so no deadlock

Reply:
1.it should not be deadlock
2.if we add default, prefetchMsg could be dropped once the channel is full.

Reply:
make sense.